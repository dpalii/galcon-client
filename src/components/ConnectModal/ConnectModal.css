.backdrop {
    top: 0;
    left: 0;
    position: fixed;
    width: 100vw;
    height: 100vh;
    background-color: rgba(0, 0, 0, 0.5);
    z-index: 1;
}

.form {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    background-color: var(--bg-primary);
    border-radius: 8px;
    padding: 20px;
}

.form form {
    margin-bottom: 20px;
}

.form legend {
    padding-bottom: 20px;
    font-size: 1.2rem;
    font-weight: bold;
}

.form input {
    border: none;
    background-color: var(--bg-secondary);
    padding: 10px 15px;
    border-radius: 4px 0 0 4px;
}

.form input:focus {
    outline: none;
}

.form form button {
    color: var(--text-success);
    border: none;
    border-radius: 0 4px 4px 0;
    padding: 10px 15px;
    background-color: var(--bg-success);
    font-weight: bold;
    transition: background-color 0.3s;
    cursor: pointer;
}

.form form button:hover {
    background-color: var(--bg-success-dark);
}

.form form button:active {
    background-color: var(--bg-success-darker);
}

.form form button:focus {
    outline: none;
}

.create-new {
    color: var(--text-default);
    border: none;
    border-radius: 4px;
    padding: 10px 15px;
    background-color: var(--bg-default);
    font-weight: bold;
    transition: background-color 0.3s;
    cursor: pointer;
}

.create-new:hover {
    background-color: var(--bg-default-dark);
}

.create-new:active {
    background-color: var(--bg-default-darker);
}

.create-new:focus {
    outline: none;
}