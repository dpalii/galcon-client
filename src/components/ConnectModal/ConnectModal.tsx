import './ConnectModal.css'
import { FormEvent, useState } from 'react'

export interface ConnectModalProps {
    onConnect: (gameId: string) => void
    onClose: () => void
}

export function ConnectModal({ onConnect, onClose }: ConnectModalProps) {
    const [gameId, setGameId] = useState('')

    const onClick = (e: FormEvent) => {
        e.preventDefault()
        onConnect(gameId)
    }

    return (
        <div className="backdrop">
            <div className="modal">
                <h2 className="heading">ENTER GAME ID:</h2>
                <form className='connect-form'>
                    <input
                        value={gameId}
                        onChange={(e) => setGameId(e.target.value)}
                    />
                    <button id="submit" type="submit" onClick={onClick}>
                        JOIN
                    </button>
                </form>
                <button className="btn w-100" onClick={onClose}>
                    BACK
                </button>
            </div>
        </div>
    )
}
