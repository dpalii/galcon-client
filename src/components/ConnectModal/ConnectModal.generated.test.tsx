import React from 'react';
import { fireEvent } from '@testing-library/react';
import { render, unmountComponentAtNode } from 'react-dom';

import { act } from 'react-dom/test-utils';
import { ConnectModal } from '../../../src/components/ConnectModal/ConnectModal';

let container: any = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('<ConnectModal>', () => {
  it('should render component', () => {
    act(() => {
      render(<ConnectModal
        onConnect={(gameId: string) => {}}
        onClose={() => {}}
      />, container);
    });
    container.querySelector('.btn').dispatchEvent(new MouseEvent('click', { bubbles: true }));
    container.querySelector('#submit').dispatchEvent(new MouseEvent('click', { bubbles: true }));
    fireEvent.change(container.querySelector('input'), { target: { value: '738gf87f...' } });
    expect(container).toMatchSnapshot();
  });
});
