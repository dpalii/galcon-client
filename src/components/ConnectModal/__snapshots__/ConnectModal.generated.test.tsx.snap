// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConnectModal> should render component 1`] = `
<div>
  <div
    class="backdrop"
  >
    <div
      class="modal"
    >
      <h2
        class="heading"
      >
        ENTER GAME ID:
      </h2>
      <form
        class="connect-form"
      >
        <input
          value="738gf87f..."
        />
        <button
          id="submit"
          type="submit"
        >
          JOIN
        </button>
      </form>
      <button
        class="btn w-100"
      >
        BACK
      </button>
    </div>
  </div>
</div>
`;
